### ISSUE TITLE
NSPanel fails to come up after update to latest ESPHome 

### ISSUE THREAD

**Issue by kunaalm** [Blackymas/NSPanel_HA_Blueprint#1693]
### TFT Version

4.1.2

### ESPHome Version

2023.12.7

### Panel Model

NSPanel US Model (Landscape)

### What is the bug?

Panel fails to come up after updating to latest ESPHome 2023.12.7

### Steps to Reproduce

I have 4 US panels that I updated to latest ESPHome, all of them failed to come up after update (i.e. panel does not connect to the network).
1. I have tried updating both via OTA and flashing over serial.
2. I can confirm that flashing an older build (using ESPHome 2023.11.6) works.
3. I have also tried a vanilla 2023.12.7 ESP32 build and can confirm that the panel comes up connects.
4. I have also tried with "PREPARE FOR FIRST USE" and can confirm that the panel comes up. But following it up with an OTA updated failed to bring up the panel(s).

**Serial output after "PREPARE FOR FIRST USE"**
-----
rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13132
load:0x40080400,len:3036
entry 0x400805e4
[I][logger:351]: Log initialized
[C][ota:473]: There have been 4 suspected unsuccessful boot attempts.
[D][esp32.preferences:114]: Saving 1 preferences to flash...
[D][esp32.preferences:143]: Saving 1 preferences to flash: 0 cached, 1 written, 0 failed
[I][app:029]: Running through setup()...
[C][esp32_ble:027]: Setting up BLE...
[C][wifi:038]: Setting up WiFi...
[C][wifi:051]: Starting WiFi...
[C][wifi:052]:   Local MAC: C8:F0:9E:EA:9C:C4
[D][wifi:061]: Loaded saved wifi settings: yezdi
[D][wifi:455]: Starting scan...
[D][esp32_ble:270]: Enabling BLE...
[D][esp32_ble_server:077]: BLE server setup successfully
[D][esp32_improv.component:077]: Creating Improv service
[D][esp-idf:000]: E (5213) wifi:
[D][esp-idf:000]: AP has neither DSSS parameter nor HT Information, drop it
[D][esp-idf:000]: 

[D][esp-idf:000]: E (5316) wifi:
[D][esp-idf:000]: AP has neither DSSS parameter nor HT Information, drop it
[D][esp-idf:000]: 

[D][wifi:470]: Found networks:
[I][wifi:514]: - 'yezdi' (7A:AC:B9:B1:88:31) [redacted]▂▄▆█
[D][wifi:515]:     Channel: 1
[D][wifi:516]:     RSSI: -51 dB
[I][wifi:514]: - 'yezdi' (6A:D7:9A:26:B9:24) [redacted]▂▄▆█
[D][wifi:515]:     Channel: 6
[D][wifi:516]:     RSSI: -55 dB
[I][wifi:514]: - 'yezdi' (6A:D7:9A:25:83:9D) [redacted]▂▄▆█
[D][wifi:515]:     Channel: 6
[D][wifi:516]:     RSSI: -60 dB
[I][wifi:514]: - 'yezdi' (D2:21:F9:2A:24:EA) [redacted]▂▄▆█
[D][wifi:515]:     Channel: 6
[D][wifi:516]:     RSSI: -74 dB
[D][wifi:519]: - 'griffin'[redacted] (74:AC:B9:B1:88:31) [redacted]▂▄▆█
[D][wifi:519]: - 'enfield'[redacted] (7E:AC:B9:B1:88:31) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (6A:D7:9A:66:B9:24) [redacted]▂▄▆█
[D][wifi:519]: - 'rajdoot'[redacted] (82:AC:B9:B1:88:31) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (86:AC:B9:B1:88:31) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (8A:AC:B9:B1:88:31) [redacted]▂▄▆█
[D][wifi:519]: - 'rajdoot'[redacted] (6A:D7:9A:45:83:9D) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (04:17:B6:5C:C3:80) [redacted]▂▄▆█
[D][wifi:519]: - 'enfield'[redacted] (6A:D7:9A:36:B9:24) [redacted]▂▄▆█
[D][wifi:519]: - 'griffin'[redacted] (68:D7:9A:46:B9:24) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (6A:D7:9A:56:B9:24) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (6A:D7:9A:65:83:9D) [redacted]▂▄▆█
[D][wifi:519]: - 'rajdoot'[redacted] (6A:D7:9A:46:B9:24) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (6A:D7:9A:55:83:9D) [redacted]▂▄▆█
[D][wifi:519]: - 'griffin'[redacted] (68:D7:9A:25:83:9D) [redacted]▂▄▆█
[D][wifi:519]: - 'enfield'[redacted] (6A:D7:9A:35:83:9D) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (D2:21:F9:6A:24:EA) [redacted]▂▄▆█
[D][wifi:519]: - 'ArnavAadhith'[redacted] (8A:9C:8E:01:A0:53) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (8A:9C:8E:01:A0:51) [redacted]▂▄▆█
[D][wifi:519]: - 'rajdoot'[redacted] (D2:21:F9:4A:24:EA) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (D2:21:F9:5A:24:EA) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (8A:9C:8E:01:A0:56) [redacted]▂▄▆█
[D][wifi:519]: - 'griffin'[redacted] (D0:21:F9:6A:24:EA) [redacted]▂▄▆█
[D][wifi:519]: - 'enfield'[redacted] (D2:21:F9:3A:24:EA) [redacted]▂▄▆█
[D][wifi:519]: - 'NETGEAR91'[redacted] (28:80:88:A7:4F:4B) [redacted]▂▄▆█
[D][wifi:519]: - 'ATTnSWRGXi'[redacted] (A8:9F:EC:96:29:90) [redacted]▂▄▆█
[D][wifi:519]: - 'nitaigaurahari'[redacted] (08:36:C9:8E:A3:A9) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (4A:BD:CE:86:FB:B2) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (72:CF:0E:FB:06:2A) [redacted]▂▄▆█
[D][wifi:519]: - 'Bradley’s Wifi'[redacted] (D8:6C:63:EF:E9:B8) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (42:CF:0E:FB:06:2A) [redacted]▂▄▆█
[D][wifi:519]: - 'Home1435'[redacted] (4A:BD:CE:86:FB:B1) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (4A:BD:CE:86:FB:B4) [redacted]▂▄▆█
[D][wifi:519]: - 'ArnavAadhith'[redacted] (52:CF:0E:FB:06:2A) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (D6:E2:CB:D1:96:D2) [redacted]▂▄▆█
[D][wifi:519]: - 'NETGEAR71'[redacted] (CC:40:D0:62:C8:38) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (CE:F3:C8:C5:18:76) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (CA:99:B2:F8:67:D9) [redacted]▂▄▆█
[D][wifi:519]: - '[LG_Oven]0fc5'[redacted] (BA:16:5F:FB:0F:C5) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (CE:F3:C8:C5:18:74) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (CE:F3:C8:C5:18:77) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (3E:82:C0:08:0B:5C) [redacted]▂▄▆█
[D][wifi:519]: - 'ArnavAadhith'[redacted] (CE:F3:C8:C5:18:71) [redacted]▂▄▆█
[D][wifi:519]: - 'Krishna'[redacted] (3E:82:C0:08:0B:59) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (3E:82:C0:08:0B:5E) [redacted]▂▄▆█
[D][wifi:519]: - 'Home Internet'[redacted] (4C:AB:F8:7C:44:0C) [redacted]▂▄▆█
[D][wifi:519]: - 'HVP_48a2e6d1aa44-Pairing'[redacted] (48:A2:E6:D1:AA:45) [redacted]▂▄▆█
[D][wifi:519]: - ''[redacted] (12:59:32:BC:FD:8F) [redacted]▂▄▆█
[I][wifi:300]: WiFi Connecting to 'yezdi'...
[I][wifi:587]: WiFi Connected!
[C][wifi:405]:   Local MAC: C8:F0:9E:EA:9C:C4
[C][wifi:410]:   SSID: 'yezdi'[redacted]
[C][wifi:411]:   IP Address: 192.168.49.33
[C][wifi:413]:   BSSID: 7A:AC:B9:B1:88:31[redacted]
[C][wifi:414]:   Hostname: 'esphome-web-ea9cc4'
[C][wifi:416]:   Signal strength: -52 dB ▂▄▆█
[C][wifi:420]:   Channel: 1
[C][wifi:421]:   Subnet: 255.255.248.0
[C][wifi:422]:   Gateway: 192.168.48.1
[C][wifi:423]:   DNS1: 192.168.48.2
[C][wifi:424]:   DNS2: 0.0.0.0
[D][wifi:596]: Disabling AP...
[C][web_server:123]: Setting up web server...
[C][ota:097]: Over-The-Air Updates:
[C][ota:098]:   Address: esphome-web.local:3232
[W][ota:107]: Last Boot was an unhandled reset, will proceed to safe mode in 6 restarts
[C][api:025]: Setting up Home Assistant API server...
[I][app:062]: setup() finished successfully!
[I][app:102]: ESPHome version 2023.12.7 compiled on Jan 17 2024, 01:15:53
[I][app:104]: Project esphome.web version 1.0
[C][wifi:573]: WiFi:
[C][wifi:405]:   Local MAC: C8:F0:9E:EA:9C:C4
[C][wifi:410]:   SSID: 'yezdi'[redacted]
[C][wifi:411]:   IP Address: 192.168.49.33
[C][wifi:413]:   BSSID: 7A:AC:B9:B1:88:31[redacted]
[C][wifi:414]:   Hostname: 'esphome-web-ea9cc4'
[C][wifi:416]:   Signal strength: -53 dB ▂▄▆█
[C][wifi:420]:   Channel: 1
[C][wifi:421]:   Subnet: 255.255.248.0
[C][wifi:422]:   Gateway: 192.168.48.1
[C][wifi:423]:   DNS1: 192.168.48.2
[C][wifi:424]:   DNS2: 0.0.0.0
[C][logger:439]: Logger:
[C][logger:440]:   Level: DEBUG
[C][logger:441]:   Log Baud Rate: 115200
[C][logger:443]:   Hardware UART: UART0
[C][esp32_ble:374]: ESP32 BLE:
[C][esp32_ble:376]:   MAC address: C8:F0:9E:EA:9C:C6
[C][esp32_ble:377]:   IO Capability: none
[C][esp32_ble_server:200]: ESP32 BLE Server:
[C][esp32_improv.component:257]: ESP32 Improv:
[C][captive_portal:088]: Captive Portal:
[C][web_server:168]: Web Server:
[C][web_server:169]:   Address: esphome-web.local:80
[C][mdns:115]: mDNS:
[C][mdns:116]:   Hostname: esphome-web-ea9cc4
[C][ota:097]: Over-The-Air Updates:
[C][ota:098]:   Address: esphome-web.local:3232
[W][ota:107]: Last Boot was an unhandled reset, will proceed to safe mode in 6 restarts
[C][api:139]: API Server:
[C][api:140]:   Address: esphome-web.local:6053
[C][api:144]:   Using noise encryption: NO
[C][improv_serial:032]: Improv Serial:
[D][api:102]: Accepted 192.168.48.12
[D][api.connection:1121]: Home Assistant 2024.1.3 (192.168.48.12): Connected successfully

**Build / install output**
-----

INFO ESPHome 2023.12.7
INFO Reading configuration /config/esphome/nspanel-guestroom.yaml...
INFO Updating https://github.com/Blackymas/NSPanel_HA_Blueprint@main
INFO Updating https://github.com/esphome/esphome.git@pull/6051/head
INFO Detected timezone 'America/Chicago'
INFO Generating C++ source...
INFO Compiling app...
Processing nspanel-guestroom (board: esp32dev; framework: arduino; platform: platformio/espressif32@5.4.0)
--------------------------------------------------------------------------------
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
Dependency Graph
|-- AsyncTCP-esphome @ 2.0.1
|-- WiFi @ 2.0.0
|-- FS @ 2.0.0
|-- Update @ 2.0.0
|-- ESPAsyncWebServer-esphome @ 3.1.0
|-- ESPmDNS @ 2.0.0
|-- ArduinoJson @ 6.18.5
|-- WiFiClientSecure @ 2.0.0
|-- HTTPClient @ 2.0.0
Compiling .pioenvs/nspanel-guestroom/src/main.cpp.o
Linking .pioenvs/nspanel-guestroom/firmware.elf
RAM:   [=         ]  14.2% (used 46404 bytes from 327680 bytes)
Flash: [=======   ]  72.8% (used 1335353 bytes from 1835008 bytes)
Building .pioenvs/nspanel-guestroom/firmware.bin
Creating esp32 image...
Successfully created esp32 image.
esp32_create_combined_bin([".pioenvs/nspanel-guestroom/firmware.bin"], [".pioenvs/nspanel-guestroom/firmware.elf"])
Wrote 0x1576c0 bytes to file /data/build/nspanel-guestroom/.pioenvs/nspanel-guestroom/firmware-factory.bin, ready to flash to offset 0x0
========================= [SUCCESS] Took 86.75 seconds =========================
INFO Successfully compiled program.
INFO Connecting to 192.168.49.33
INFO Uploading /data/build/nspanel-guestroom/.pioenvs/nspanel-guestroom/firmware.bin (1341120 bytes)
Uploading: [============================================================] 100% Done...

INFO Upload took 19.74 seconds, waiting for result...
INFO OTA successful
INFO Successfully uploaded program.
INFO Starting log output from 192.168.49.33 using esphome API
WARNING Can't connect to ESPHome API for nspanel-guestroom @ 192.168.49.33: Timeout while connecting to [AddrInfo(family=<AddressFamily.AF_INET: 2>, type=<SocketKind.SOCK_STREAM: 1>, proto=6, sockaddr=IPv4Sockaddr(address='192.168.49.33', port=6053))] (TimeoutAPIError)
INFO Trying to connect to nspanel-guestroom @ 192.168.49.33 in the background

**Serial output after OTA**
-----
rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13132
load:0x40080400,len:3036
entry 0x400805e4

**Serial output after reset (grounding RST pad)**
-----
bkcmd=0���sleep=0���bkcmd=0���sleep=0���rest���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���boguscommand=0���DRAKJHSUYDGBNCJHGJKSHBDN���connect���bkcmd=3���dim=100���thup=1���get dp���get charset���get display_mode���page boot���get tft_version.txt���vis tft_version,1���tft_version.txt="4.1.2"���wakeup_timer.en=1���get dim���brightness=100���settings.brightslider.val=100���brightness_dim=10���settings.dimslider.val=10���brightness_sleep=0���home.icon_top_01.font=7���home.icon_top_02.font=7���home.icon_top_03.font=7���home.icon_top_04.font=7���home.icon_top_05.font=7���home.icon_top_06.font=7���home.icon_top_07.font=7���home.icon_top_08.font=7���home.icon_top_09.font=7���home.icon_top_10.font=7���home.button01.font=8���home.button02.font=8���home.button03.font=8���home.button04.font=8���home.button05.font=8���home.button06.font=8���home.button07.font=8���home.bt_notific.font=8���home.bt_qrcode.font=8���home.bt_entities.font=8���home.wifi_icon.font=7���is_notification=0���get dim���home.current_temp.txt="21.5°C"���dim=10���get dim���get dim���dim=10���get dim���get dim���home.current_temp.txt="21.7°C"���page screensaver���dim=0���get dim���get dim���home.current_temp.txt="21.5°C"���dim=10���get dim���get dim���page screensaver���dim=0���get dim���get dim���home.current_temp.txt="21.6°C"���dim=10���get dim���get dim���page screensaver���dim=0���get dim���get dim���home.current_temp.txt="21.7°C"���dim=10���get dim���get dim���page screensaver���dim=0���get dim���get dim���home.current_temp.txt="21.5°C"���dim=10���get dim���get dim���page screensaver���dim=0���get dim���get dim���home.current_temp.txt="21.6°C"���home.current_temp.txt="21.5°C"���home.current_temp.txt="21.4°C"���home.current_temp.txt="21.7°C"���home.current_temp.txt="21.5°C"���home.current_temp.txt="21.3°C"���home.current_temp.txt="21.3°C"���ets Jul 29 2019 12:21:46

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13132
load:0x40080400,len:3036
entry 0x400805e4

### Your panel's YAML

```yaml
substitutions:
  ###### CHANGE ME START ######
  device_name: "nspanel-guestroom" 
  friendly_name: "NSPanel: GuestRoom"
  wifi_ssid: !secret wifi_ssid
  wifi_password: !secret wifi_password
  ota_password: !secret ota_password

  nextion_update_url: "http://192.168.48.12:8123/local/nspanel_us.tft"

  ##### addon-configuration #####
  ## addon_climate ##
  # addon_climate_heater_relay: "1" # possible values: 1/2

  ##### CHANGE ME END #####

packages:
  remote_package:
    url: https://github.com/Blackymas/NSPanel_HA_Blueprint
    ref: main
    files:
      - nspanel_esphome.yaml # Core package
      # - nspanel_esphome_advanced.yaml # activate advanced (legacy) elements - can be useful for troubleshooting
      # - nspanel_esphome_addon_climate_cool.yaml # activate for local climate (cooling) control
      # - nspanel_esphome_addon_climate_heat.yaml # activate for local climate (heater) control
    refresh: 300s

wifi:
  manual_ip:
    static_ip: 192.168.49.33
    gateway: 192.168.1.254
    subnet: 255.255.255.0

##### My customization - Start #####
web_server:
  port: 80

##### My customization - End #####
```

**Comment by edwardtfn** [comment]
When you flash it with this project's firmware, what do you see on panel's screen?
Is the fallback AP available?

**Comment by edwardtfn** [comment]
Could you please try flashing your panel with this:
```yaml
substitutions:
  ###### CHANGE ME START ######
  device_name: "nspanel-guestroom"
  friendly_name: "NSPanel: GuestRoom"
  wifi_ssid: !secret wifi_ssid
  wifi_password: !secret wifi_password
  ota_password: !secret ota_password

  nextion_update_url: "http://192.168.48.12:8123/local/nspanel_us.tft"

  ##### addon-configuration #####
  ## addon_climate ##
  # addon_climate_heater_relay: "1" # possible values: 1/2

  ##### CHANGE ME END #####

packages:
  remote_package:
    url: https://github.com/Blackymas/NSPanel_HA_Blueprint
    ref: main
    files:
      - nspanel_esphome.yaml # Core package
      # - nspanel_esphome_advanced.yaml # activate advanced (legacy) elements - can be useful for troubleshooting
      # - nspanel_esphome_addon_climate_cool.yaml # activate for local climate (cooling) control
      # - nspanel_esphome_addon_climate_heat.yaml # activate for local climate (heater) control
    refresh: 300s

esp32:
  framework:
    type: esp-idf

##### My customization - Start #####

captive_portal:
  
web_server:

wifi:
  id: wifi_component
  power_save_mode: LIGHT
  networks:
    - id: !extend wifi_default
      ssid: !secret wifi_ssid
      password: !secret wifi_password
  ap:
    ssid: "nspanel-guestroom"
    password: !secret wifi_password
  manual_ip:
    static_ip: 192.168.49.33
    gateway: 192.168.1.254
    subnet: 255.255.255.0

##### My customization - End #####
```
My changes are:
1. I'm avoiding substitutions for the Wi-Fi credentials, to reduce chances of error. We can deep dive on this if we get one step ahead (your panel connecting to your wifi)
2. Removed `friendly_name` as this is not fully compatible with this project and can cause issues for the blueprint to find the services. You can still rename your panel on HA UI, but please don't do this for now, until we find the issue here.
3. Changed framework to `esp-idf`. As you are flashing via serial, this is a good chance to change it. This will be default from March.
4. Added `captive_portal` to help with the troubleshooting.

**Comment by kunaalm** [comment]
Thanks, I tried flashing with the modified yaml. It still doesn't come up. I don't it see it fall back to the captivate portal either. I flashed it twice with the same result, the second time I upgraded to ESPHome 2023.12.8 (latest) 

**Log from serial port on reset**

-----
rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:6608
load:0x40078000,len:15060
ho 0 tail 12 room 4
load:0x40080400,len:3816
entry 0x40080698
I (29) boot: ESP-IDF 4.4.5 2nd stage bootloader
I (29) boot: compile time 10:43:43
I (29) boot: chip revision: v3.0
I (32) boot.esp32: SPI Speed      : 40MHz
I (37) boot.esp32: SPI Mode       : DIO
I (41) boot.esp32: SPI Flash Size : 4MB
I (46) boot: Enabling RNG early entropy source...
I (51) boot: Partition Table:
I (55) boot: ## Label            Usage          Type ST Offset   Length
I (62) boot:  0 otadata          OTA data         01 00 00009000 00002000
I (70) boot:  1 phy_init         RF data          01 01 0000b000 00001000
I (77) boot:  2 app0             OTA app          00 10 00010000 001c0000
I (85) boot:  3 app1             OTA app          00 11 001d0000 001c0000
I (92) boot:  4 nvs              WiFi data        01 02 00390000 0006d000
I (100) boot: End of partition table
I (104) boot: No factory image, trying OTA 0
I (109) esp_image: segment 0: paddr=00010020 vaddr=3f400020 size=2a30ch (172812) map
I (180) esp_image: segment 1: paddr=0003a334 vaddr=3ffb0000 size=02f70h ( 12144) load
I (185) esp_image: segment 2: paddr=0003d2ac vaddr=40080000 size=02d6ch ( 11628) load
I (190) esp_image: segment 3: paddr=00040020 vaddr=400d0020 size=de294h (909972) map
I (523) esp_image: segment 4: paddr=0011e2bc vaddr=40082d6c size=11700h ( 71424) load
I (563) boot: Loaded app from partition at offset 0x10000
I (627) boot: Set actual ota_seq=1 in otadata[0]
I (627) boot: Disabling RNG early entropy source...
I (638) cpu_start: Pro cpu up.
I (638) cpu_start: Starting app cpu, entry point is 0x40081e64
I (0) cpu_start: App cpu up.
I (654) cpu_start: Pro cpu start user code
I (654) cpu_start: cpu freq: 160000000
I (654) cpu_start: Application information:
I (659) cpu_start: Project name:     nspanel-guestroom
I (665) cpu_start: App version:      2023.12.8
I (670) cpu_start: Compile time:     Jan 19 2024 10:42:27
I (676) cpu_start: ELF file SHA256:  97cb60180b653001...
I (682) cpu_start: ESP-IDF:          4.4.5
I (687) cpu_start: Min chip rev:     v0.0
I (691) cpu_start: Max chip rev:     v3.99 
I (696) cpu_start: Chip rev:         v3.0
I (701) heap_init: Initializing. RAM available for dynamic allocation:
I (708) heap_init: At 3FFAE6E0 len 00001920 (6 KiB): DRAM
I (714) heap_init: At 3FFB8270 len 00027D90 (159 KiB): DRAM
I (720) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (727) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM
I (733) heap_init: At 4009446C len 0000BB94 (46 KiB): IRAM
I (741) spi_flash: detected chip: generic
I (744) spi_flash: flash io: dio
I (750) cpu_start: Starting scheduler on PRO CPU.
I (0) cpu_start: Starting scheduler on APP CPU.

**Comment by kunaalm** [comment]
![IMG_6303](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/9190016/4592f8a5-d36b-410a-8af9-57013588e70f)

**Comment by arkoulikosta** [comment]
i had the same issue, found the 4.1.2 yaml and the display loaded correctly, then uploaded the tft file, then loaded the 4.2.4 yaml and it worked

**Comment by kunaalm** [comment]
@arkoulikosta, I don't think its a problem with the TFT. One should able to connect to the device via the web portal, which I can't.
 
@edwardtfn, In from the wifi router I can see that the device is connected to the network, but the device does not respond to a ping.

**Comment by edwardtfn** [comment]
What if you install another firmware in this device? I mean, create a new device from ESPHome Dashboard, select ESP32, follow the steps there and flash your device with that basic firmware.
Would that work?

**Comment by edwardtfn** [comment]
@kunaalm, any update to this?
Are you still having those issues?

**Comment by kunaalm** [comment]
@edwardtfn, I was out traveling this week. I was able to do a scratch build and bring up the panels on DHCP. I did some debugging on my network today and realized that there was a problem with my router. I can confirm that the panel comes up, I am still using the arduino framework. I will try with IDF later. I have updated the Blueprint and TFT to 2.4.2 and everything is working. Thanks for your patience and help!

**Comment by edwardtfn** [comment]
I've added support to Firendly Name on `dev` and that will be included in the next release.

**Comment by edwardtfn** [comment]
We just released a patch with solves this: [v4.2.5 - Celebrating 1000 Stars with Enhanced Functionality and UI Improvements](https://github.com/Blackymas/NSPanel_HA_Blueprint/releases/tag/v4.2.5)
Could you please update your system and let me know if the issue was solved?
